src/types.rs

```diff
@@ -303,34 +325,44 @@ impl Id {
       public_keys : (pub_sign_key, pub_asym_key),
       secret_keys : (sec_sign_key, sec_asym_key),
       validation_token : validation_token,
-      name : NameType::new(digest.0)
+      name : NameType::new(digest.0),
+      original_name : NameType::new(digest.0),
```

Comment:
Whops, Id isn't serialisable, but the PublicId is. Still would be 64B wasted in memory if we don't need it.